src/rt/gtfsrt_update.cc

```diff
@@ -419,11 +441,149 @@ bool update_run(source_idx_t const src,
+                             date::sys_seconds message_time,
+                             std::shared_ptr<opentelemetry::trace::Span>& span,
+                             statistics& stats) {
+  // check for duplicate timestamps
+  hash_map<uint64_t, int> counts;
+  for (auto const& ent : entities) {
+    ++counts[ent.vehicle().timestamp()];
+  }
+  vector<gtfsrt::FeedEntity> accepted_entities;
+  for (auto const& ent2 : entities) {
+    if (counts[ent2.vehicle().timestamp()] == 1) {
+      accepted_entities.emplace_back(ent2);
+    } else {
+      ++stats.duplicate_vehicle_position_timestamps_;
+    }
+  }
+  // handle accepted entities
+  for (auto const& entity : accepted_entities) {
+    try {
+      auto const vp = entity.vehicle();
+      auto const vp_lat = vp.position().latitude();
+      auto const vp_lon = vp.position().longitude();
+      auto const& td = vp.trip();
+      auto const& gtfsrt_trip_id = td.trip_id();
+
+      auto [r, trip_idx] = gtfsrt_resolve_run(today, tt, &rtt, src, td,
+                                              std::string_view{gtfsrt_trip_id});
+      if (!r.valid()) {
+        log(log_lvl::error, "rt.gtfs.unsupported",
+            R"(Run could not be resolved. Skipping Message)", tag, entity.id());
+        continue;
+      }
+
+      auto const& rtt_const = rtt;
+      auto const location_seq =
+          r.is_scheduled()
+              ? std::span{tt.route_location_seq_
+                              [tt.transport_route_[r.t_.t_idx_]]}
+              : std::span{rtt_const.rt_transport_location_seq_[r.rt_]};
+
+      // add rt_transport if not existent
+      if (!r.is_rt()) {
+        r.rt_ = rtt.add_rt_transport(src, tt, r.t_);
+      }
+
+      // match position to stop
+      auto const stop_it = utl::find_if(location_seq, [&](auto const& loc) {
+        auto const stop_position =
+            tt.locations_.get(location_idx_t{loc & 0x0FFFFFFF});
+        auto const vehicle_position = geo::latlng{vp_lat, vp_lon};
+        return geo::approx_squared_distance(
+                   stop_position.pos_, vehicle_position,
+                   geo::approx_distance_lng_degrees(vehicle_position)) < 10;
+      });
+      if (stop_it == end(location_seq)) {
+        log(log_lvl::info, "rt.gtfs.unsupported",
+            R"(Position of Vehicle was not near stop. Skipping Message)", tag,
+            entity.id());
+        ++stats.vehicle_position_position_not_at_stop_;
+        continue;
+      }
+
+      // check for outdated or illegal timestamp
+      auto const vp_ts_utt = unixtime_t{std::chrono::duration_cast<i32_minutes>(
+          std::chrono::seconds{vp.timestamp()})};
+      if ((rtt.rt_transport_latest_vehicle_time_.contains(r.rt_)
+               ? rtt.rt_transport_latest_vehicle_time_[r.rt_] > vp_ts_utt
+               : tt.event_time(r.t_, 0, event_type::kDep) > vp_ts_utt) ||
+          vp_ts_utt > std::chrono::time_point_cast<i32_minutes>(
+                          std::chrono::system_clock::now())) {
```

Comment:
performance: don't call `now` all the time - we can assume it's constant for one update run